### PR TITLE
Check that client state is zeroed out for ibc client upgrade proposals

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #291 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by charleenfei — APPROVED** [review]
same comment as previous PR otherwise but otherwise nice one

File: modules/core/02-client/types/proposal.go [hunk 1]
```diff
@@ -2,11 +2,11 @@ package types
 
 import (
 	"fmt"
-
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"reflect"
```
Comment (damiannolan) [inline modules/core/02-client/types/proposal.go:9]: nit: group import statements like so:

```go
import (
        // standard library imports
	"fmt"
	"testing"
        
        // external library imports
	"github.com/stretchr/testify/require"
	abci "github.com/tendermint/tendermint/abci/types"
        
         // ibc-go library imports
	"github.com/cosmos/ibc-go/modules/core/23-commitment/types"
)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -81,6 +81,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (modules/core/04-channel) [\#1130](https://github.com/cosmos/ibc-go/pull/1130) Call `packet.GetSequence()` rather than passing func in `WriteAcknowledgement` log output
 * (apps/29-fee) [\#1278](https://github.com/cosmos/ibc-go/pull/1278) The URI path for the query to get all incentivized packets for a specific channel did not follow the same format as the rest of queries.
 * (apps/29-fee)[\#1343](https://github.com/cosmos/ibc-go/pull/1523) Fixed an issue where a bad refund address would prevent channel closure.
+* (modules/core/02-client)[\#1529](https://github.com/cosmos/ibc-go/pull/1529) Client state must be zeroed out for `UpgradeProposals` to pass validation.
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply CHANGELOG.md:84]: I think it would be clearer if we refer to `ClientState` as a type name without space

**Review by damiannolan — COMMENTED** [review]
I think we can target this for the `02-client-refactor` branch based off [this comment](https://github.com/cosmos/ibc-go/issues/292#issuecomment-1129883474)

Thanks @chatton!

**Review by crodriguezvega — APPROVED** [review]
I agree with @charleenfei's and @damiannolan's comments. Other than that, nice work, @chatton!

**Comment by chatton** [comment]
Closing in favour of #1676

**Pull request closed without merging**